### PR TITLE
Add totals row to new HSR

### PR DESCRIPTION
**Pull request by esoergel**
http://manage.dimagi.com/default.asp?153475 (not that it provides context or anything)

### CODE REVIEW

File: custom/opm/reports.py [hunk 1]
```diff
@@ -955,14 +955,29 @@ def awc_data(self):
 
     @property
     def rows(self):
+        totals = [[None, None] for i in range(len(self.model.method_map))]
+        def add_to_totals(col, val, denom):
```
Comment (dimagimon) [inline custom/opm/reports.py]: E301 expected 1 blank line, found 0

File: custom/opm/reports.py [hunk 2]
```diff
@@ -955,14 +955,29 @@ def awc_data(self):
...
+                    total = totals[col][i]
+                    totals[col][i] = total + num if total is not None else num
+
+        rows = []
         for awc in map(AWCHealthStatus, self.awc_data().values()):
-            yield [self.format_cell(getattr(awc, method),
-                                    getattr(awc, denom))
-                   for method, _, _, denom in self.model.method_map]
+            row = []
+            for col, (method, _, _, denom) in enumerate(self.model.method_map):
```
Comment (dimagimon) [inline custom/opm/reports.py]: F402 import '_' from line 20 shadowed by loop variable

File: custom/opm/reports.py [hunk 3]
```diff
@@ -955,14 +955,29 @@ def awc_data(self):
...
+                val = getattr(awc, method)
+                denominator = getattr(awc, denom)
+                row.append(self.format_cell(val, denominator))
+                add_to_totals(col, val, denominator)
+            rows.append(row)
+
+        self.total_row = [self.format_cell(val, denom) for val, denom in totals]
```
Comment (dimagimon) [inline custom/opm/reports.py]: F812 list comprehension redefines 'val' from line 969

File: custom/opm/reports.py [hunk 4]
```diff
@@ -955,14 +955,30 @@ def awc_data(self):
 
     @property
     def rows(self):
+
+        totals = [[None, None] for i in range(len(self.model.method_map))]
+        def add_to_totals(col, val, denom):
```
Comment (dimagimon) [inline custom/opm/reports.py]: E301 expected 1 blank line, found 0

Comment (esoergel) [inline custom/opm/reports.py]: I disagree.  This function exists to mutate `totals` - I find it clearer this way.

**Pull request merged**